nifty-ssl/src/main/java/com/facebook/nifty/ssl/ThreadLocalSslBufferPool.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2012-2016 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.nifty.ssl;
+
+import org.jboss.netty.handler.ssl.SslBufferPool;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import static com.facebook.nifty.core.NettyConfigBuilderBase.DEFAULT_WORKER_THREAD_COUNT;
+
+public class ThreadLocalSslBufferPool extends SslBufferPool {
+    private final ThreadLocal<NonBlockingSslBufferPool> pool;
+
+    public ThreadLocalSslBufferPool(int maxPoolSize, boolean preallocate, boolean allocateDirect) {
+        this(maxPoolSize, preallocate, allocateDirect, DEFAULT_WORKER_THREAD_COUNT);
```

Comment:
should we pass a 1 size buffer without preallocation? otherwise the superclass will allocate in its ctor